src/lib/process/index.ts

```diff
@@ -50,29 +52,44 @@ export const createProcess = async (
   dryRun: boolean = false,
   options: Polkadot.Options = defaultOptions
 ): Promise<Process.Result> => {
-  const program: Process.Program = validate(userProgram)
-  const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
-  const polkadot: Polkadot.Polkadot = await createNodeApi(options)
-  const expectedVersion: number = (await getVersion(polkadot, processId)) + 1
+  try {
 
-  if (version !== expectedVersion) {
-    throw new VersionError(`Version: ${version} must be incremented to: ${expectedVersion}`)
-  }
+    const program: Process.Program = validate(userProgram)
+    const processId = utf8ToHex(name, Constants.PROCESS_ID_LENGTH)
+    const polkadot: Polkadot.Polkadot = await createNodeApi(options)
+    const currentVersion: number = await getVersion(polkadot, processId)
+    const expectedVersion: number = currentVersion + 1
 
-  if (dryRun)
-    return {
-      process: null,
-      message: 'Dry run: transaction has not been created',
-      name,
-      version: expectedVersion,
-      program,
+    if (version > expectedVersion || version < currentVersion)
+      throw new VersionError(version, expectedVersion, name)
+
+    if (version === currentVersion) {
+      const process = await getProcess(polkadot, processId, version)
+
+      if (program.length !== process.program.length) 
+        throw new ProgramError('existing: programs are different lengths', process) 
+
+      if (!program.every((step, i) => textify(step) === textify(process.program[i])))
+        throw new ProgramError('existing: program steps did not match', process) 
+
+      throw new ProgramError(`Process ${name} is already created.`, process)
     }
 
-  const process: Process.Payload = await createProcessTransaction(polkadot, processId, program, options)
+    if (dryRun)
+      return {
+        process: null,
+        message: 'Dry run: transaction has not been created',
+        name,
+        version: expectedVersion,
+        program,
+      }
 
-  return {
-    message: `Transaction for new process ${name} has been successfully submitted`,
-    process,
+    return {
+      message: `Transaction for new process ${name} has been successfully submitted`,
+      process: await createProcessTransaction(polkadot, processId, program, options),
+    }
+  } catch (err: any) {
+    return err
   }
```

Comment:
this is maybe a touch odd. The return type is ` Promise<Process.Result>` but we're returning an `any`? I'm surprised the compiler is ok with that. Maybe we should probably be catching the error at the next level up?

Reply:
this is because the way try/catch works. The typescript impementation would be either way. Removed all together after looking at other people implementations. It won't check catch block